strawberryfields/plot.py

```diff
@@ -120,3 +123,236 @@ def generate_wigner_chart(data, xvec, pvec, contours=True):
     chart["layout"]["scene"]["zaxis"]["title"] = "W(x,p)"
 
     return chart
+
+
+# Plot.ly default barchart JSON
+barchart_default = {
+    "data": [{"y": [], "x": [], "type": "bar", "name": "q[0]"}],
+    "layout": {
+        "width": 835,
+        "height": 500,
+        "margin": {"l": 100, "r": 100, "b": 100, "t": 100, "pad": 4},
+        "paper_bgcolor": "rgba(0,0,0,0)",
+        "plot_bgcolor": "rgba(0,0,0,0)",
+        "autosize": True,
+        "yaxis": {
+            "gridcolor": "#bbb",
+            "type": "linear",
+            "autorange": True,
+            "title": "Probability",
+            "fixedrange": True,
+        },
+        "xaxis": {
+            "gridcolor": textcolor,
+            "type": "category",
+            "autorange": True,
+            "title": "q[0]",
+            "fixedrange": True,
+        },
+        "showlegend": False,
+        "annotations": [
+            {
+                "showarrow": False,
+                "yanchor": "bottom",
+                "xref": "paper",
+                "xanchor": "center",
+                "yref": "paper",
+                "text": "",
+                "y": 1,
+                "x": 0,
+                "font": {"size": 16},
+            }
+        ],
+    },
+    "config": {
+        "modeBarButtonsToRemove": ["zoom2d", "lasso2d", "select2d", "toggleSpikelines"],
+        "displaylogo": False,
+    },
+}
+
+
+def plot_fock(state, modes, cutoff=None, renderer="browser"):
+    """Plot the Fock state probabilities with Plot.ly.
+
+    Args:
+        state (:class:`.BaseState`): the state used for plotting
+        modes (list): list of modes to generate output for
+        cutoff (int): the cutoff value determining the maximum Fock state to
+            get probabilities for, required if state is a
+            :class:`~.BaseGaussianState`
+        renderer (string): the renderer for plotting with Plot.ly
+    """
+    if cutoff is None:
+        if isinstance(state, BaseGaussianState):
+            raise ValueError(f"No cutoff specified for {state.__class__}.")
+
+        # BaseFockState has cutoff
+        cutoff = state.cutoff_dim
+
+    pio = _get_plotly()
+    pio.renderers.default = renderer
+
+    num_modes = len(modes)
+
+    # Reduced density matrices
+    rho = [state.reduced_dm(n, cutoff=cutoff) for n in range(num_modes)]
+    photon_dists = np.array([np.real(np.diagonal(p)) for p in rho])
+
+    n = np.arange(cutoff)
+    mean = [np.sum(n * probs).real for probs in photon_dists]
+
+    xlabels = [fr"$|{i}\rangle$" for i in range(0, cutoff, 1)]
+
+    new_chart = generate_fock_chart(modes, photon_dists, mean, xlabels)
+    pio.show(new_chart)
+
+
+def generate_fock_chart(modes, photon_dists, mean, xlabels):
```

Comment:
@antalszava I think we should pass the state to _this_ function, and do the photon dist computation here, instead.

I just suddenly realised that if running SF on a server (e.g., simulon), there is no function that can be called, given a state, to compute the JSON plot data.

Reply:
Great point! Thanks for further clarifying. Updated all 3 plotting functions like that.